### ISSUE TITLE
DE algorithm signature

### ISSUE THREAD

**Issue by half1red** [esa/pagmo2#182]
I have a weird error when using pygmo.de().
If I specify xtol, I get an error telling me that the signature if not correct (see below). The same code works with tol instead of xcol. However, in the source code I have checked that xtol is the right signature.

I am using pygmo installed through pip, in latest version (2.7) and python 3.6.3.
```
Traceback (most recent call last):
  File "[...]/test.py", line 129, in <module>
    pg.de(gen=10000, F=0.5, CR=0.3, variant=9, ftol=1e-3, xtol=1e-2))
Boost.Python.ArgumentError: Python argument types in
    de.__init__(de)
did not match C++ signature:
    __init__(_object*, unsigned int gen=1, double F=0.8, double CR=0.9, unsigned int variant=2, double ftol=1e-06, double tol=1e-06, unsigned int seed)
    __init__(_object*, unsigned int gen=1, double F=0.8, double CR=0.9, unsigned int variant=2, double ftol=1e-06, double tol=1e-06)
    __init__(_object*)

```

**Comment by bluescarni** [comment]
I can confirm that the ``xtol`` argument has been exposed as ``tol`` in the Python bindings code:

https://github.com/esa/pagmo2/blob/master/pygmo/expose_algorithms.cpp#L286

Incidentally, it also seems like the ``ftol``/``xtol`` parameters' descriptions in the docs have been swapped around, both C++ and Python:

https://esa.github.io/pagmo2/docs/cpp/algorithms/de.html#_CPPv2N5pagmo2deE
https://esa.github.io/pagmo2/docs/python/algorithms/py_algorithms.html#pygmo.de

@darioizzo what is the correct naming scheme here?

**Comment by bluescarni** [comment]
I have a PR with a tentative fix up at #183.

**Comment by darioizzo** [comment]
@bluescarni Thanks!, it seem indeed this slipped in the code. I guess since de was coded very early (i think was the first UDA) on the python interface tests were not done ... Your PR seems to address correctly the issue.

**Comment by bluescarni** [comment]
This should now be fixed in master via #183.
